File: dimik/dimik-72.cpp
#include <iostream>
#include <cmath>
using namespace std;
int main()
{
    int repeat, i, a,b,c;
    float s, area;
    cout << fixed;
    cout.precision(2);
    cin >> repeat;
    for(i=0; i<repeat;i++)
        {
            cin >> a >> b >> c;
            s = (a + b + c)/2.0;
            area = sqrt(s*(s-a)*(s-b)*(s-c));
            if(area >= 0.0)
            {

                cout << area << endl;
            }
            else
            {
                 cout << "Oh, No!" << endl;
            }
        }
    return 0;
}

File: dimik/dimik-72.py
from math import sqrt
repeat = int(input())
for x in range(repeat):
    a, b, c = map(int, input().split())
    try:
        s = (a + b + c) / 2
        area = sqrt(s * (s - a) * (s - b) * (s - c))
        print("%.2f" % area)
    except ValueError:
        print('Oh, No!')
